tools/wandb_export_history.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+"""Export W&B run history.
+
+This module uses the W&B public api to download unsampled history to a
+sqlite database.
+
+Usage:
+    ./wandb_export_history --run entity/project/run_id --db_file save.db
+
+or:
+    ```python
+    import wandb_export_history
+    wandb_export_history(run="entity/project/run_id")
+    ```
+
+"""
+
+import argparse
+import sqlite3
+from typing import Dict, Iterator, List
+
+import pandas as pd  # type: ignore
+import wandb
+
+
+DB_FILE = "run.db"
+
+
+def chunk(n: int, iterable) -> Iterator[List[Dict]]:
+    done = False
+    while not done:
+        data = []
+        try:
+            for _ in range(n):
+                data.append(next(iterable))
+        except StopIteration:
+            if data:
+                yield data
+            # done = True
+            break
+        yield data
+
+
+def wandb_export_history(
+    *,
+    run,
+    api=None,
+    db_file=None,
+    db_table=None,
+    db_replace: bool = None,
+    history_exclude_prefix=None,
+    read_page_size=None,
+    write_page_size=None,
```

Comment:
nit: why not specify the default as part of arguments?

Reply:
Because then there would be default args in both argparse and the function definition.   The way it is written now, you can use this as a command line or import it and use it as a function.

Reply:
no i meant instead of Nones, you assign defaults in the body of the function, why not do it as part of the function arguments